### ISSUE TITLE
Add support to install plugins from branch names with slashes

### ISSUE THREAD

**Issue by joseluisq** [jorgebucaran/fisher#588]
It's not possible to install a plugins from a branch that uses for example [Gitflow](https://github.com/nvie/gitflow) conventions due that branch names can contain slashes and the `_fisher_fetch` function because is [splinting a string based on slashes](https://github.com/jorgebucaran/fisher/blob/c2c091f33ea83cbcb67ac2d8d912c2cfddd5e4a6/fisher.fish#L294-L301) as well, throws an error when is trying to fetch the tarball.

`fisher 3.3.0`

```sh
~> fisher add username/plugin@feature/some-feature
fetching username/plugin
fisher: cannot add username/plugin -- is this a valid plugin?
```

**Comment by jorgebucaran** [comment]
@joseluisq Maybe we can rewrite that part and come up with an alternative that uses less code and supports this particular use case. I'll keep this on my radar. Thanks.

**Comment by jorgebucaran** [comment]
Should be fixed in Fisher 4. Landing this week. 👍

**Comment by jorgebucaran** [comment]
@joseluisq Actually you can install a plugin@foo/bar/baz now with 4.x, but `update` and `remove` break apart. Handling `/` in branch names is a lot more involved than I expected. Do you want to give it a shot?

**Comment by jorgebucaran** [comment]
Implemented in [`4.1.0`](https://github.com/jorgebucaran/fisher/releases/tag/4.1.0). 🎉
